docs/JAX_examples_image_segmentation.md

```diff
@@ -0,0 +1,1389 @@
+    data_source=train_dataset,
+    sampler=train_sampler,                 # Sampler to determine how to access the data
+    worker_count=4,                        # Number of child processes launched to parallelize the transformations among
+    worker_buffer_size=2,                  # Count of output batches to produce in advance per worker
+    operations=[
+        DataAugs(train_transforms),
+        grain.Batch(train_batch_size, drop_remainder=True),
+    ]
+)
+
+# Validation dataset loader
+val_loader = grain.DataLoader(
+    data_source=val_dataset,
+    sampler=val_sampler,                   # Sampler to determine how to access the data
+    worker_count=4,                        # Number of child processes launched to parallelize the transformations among
+    worker_buffer_size=2,
+    operations=[
+        DataAugs(val_transforms),
+        grain.Batch(val_batch_size),
+    ]
+)
+
+# Training dataset loader for evaluation (without dataaugs)
+train_eval_loader = grain.DataLoader(
+    data_source=train_dataset,
+    sampler=train_sampler,                 # Sampler to determine how to access the data
+    worker_count=4,                        # Number of child processes launched to parallelize the transformations among
+    worker_buffer_size=2,                  # Count of output batches to produce in advance per worker
+    operations=[
+        DataAugs(val_transforms),
+        grain.Batch(val_batch_size),
+    ]
+)
+```
+
+```{code-cell} ipython3
+train_batch = next(iter(train_loader))
+val_batch = next(iter(val_loader))
+```
+
+```{code-cell} ipython3
+print("Train images batch info:", type(train_batch["image"]), train_batch["image"].shape, train_batch["image"].dtype)
+print("Train masks batch info:", type(train_batch["mask"]), train_batch["mask"].shape, train_batch["mask"].dtype)
+```
+
+Let's display the training and validation data
+
+```{code-cell} ipython3
+images, masks = train_batch["image"], train_batch["mask"]
+
+for img, mask in zip(images[:3], masks[:3]):
+    display_datapoint({"image": img, "mask": mask}, label=" (augmented train set)")
+```
+
+```{code-cell} ipython3
+images, masks = val_batch["image"], val_batch["mask"]
+
+for img, mask in zip(images[:3], masks[:3]):
+    display_datapoint({"image": img, "mask": mask}, label=" (augmented validation set)")
+```
+
+## Model for Image Segmentation
+
+In this section we will implement from scratch [UNETR](https://arxiv.org/abs/2103.10504) model using Flax NNX. Reference PyTorch implementation of this model can be found on the GitHub in [MONAI library](https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/nets/unetr.py).
+
+The UNETR model utilizes a transformer as the encoder to learn sequence representations of the input and to capture the global multi-scale information, while also following the “U-shaped” network design like [UNet](https://arxiv.org/abs/1505.04597) model:
+![image.png](attachment:ed48cb36-b8c8-442f-b626-28e403b61ca2.png)
```

Comment:
Did you mean to include this image in the PR?

Reply:
It looks like notebook attachments are not supported: https://github.com/executablebooks/MyST-NB/issues/230

Maybe we can put it separately in an `html_static_path`